fpdf/fpdf.py

```diff
@@ -2461,15 +2520,28 @@ def link(self, x, y, w, h, link, alt_text=None, **kwargs):
             y (float): vertical position (from the top) to the bottom side of the link rectangle
             w (float): width of the link rectangle
             h (float): height of the link rectangle
-            link: either an URL or an integer returned by `FPDF.add_link`, defining an internal link to a page
+            link: can be one of the following:
+                - a URL string to create an external link
+                - an integer returned by `FPDF.add_link`, defining an internal link to a page
+                - a named destination string prefixed with '#' (e.g., '#chapter1')
             alt_text (str): optional textual description of the link, for accessibility purposes
             border_width (int): thickness of an optional black border surrounding the link.
                 Not all PDF readers honor this: Acrobat renders it but not Sumatra.
         """
         action, dest = None, None
         if link:
             if isinstance(link, str):
-                action = URIAction(link)
+                # Check if this is a named destination (prefixed with '#')
+                if link.startswith("#"):
+                    dest_name = link[1:]  # Remove the '#' prefix
+                    if dest_name in self.named_destinations:
+                        # Use destination name instead of destination object for named destinations
+                        dest = PDFString(dest_name, encrypt=True)
+                    else:
+                        raise KeyError(f"Named destination '{dest_name}' not found")
```

Comment:
```suggestion
                    if dest_name not in self.named_destinations:
                        self.named_destinations[dest_name] = DestinationXYZ(0, top=self.h_pt * self.k)
                    dest = PDFString(dest_name, encrypt=self._security_handler)
```

This will allow you to use a named destination without reserving the name before
The named link can be set after and it will work
If the link is never set, the default is to sent to the top of the first page